### ISSUE TITLE
Improve accessibility on Switch component

### ISSUE THREAD

**Issue by soykje** [adevinta/spark#796]
### Description

With current DOM structure (label tag wrapping the whole content) we recently noticed a small accessibility concern, about accessible name.

Indeed, the component itself, identified through the `role="switch"` attribute, doesn't have any accessible name. This is due to component text (aka children) added as a sibling DOM node, and so not directly related. This can be confirmed using the "Axe" Chrome accessibility extension, and testing the component.

After discussions with @andresz1, @acd02 and @Powerplex we agreed:
- changing the DOM structure, to match the Radix pattern: having the label as a sibling, linked through an `htmlFor` attribute. The whole content wrapped within a neutral div, for ex.
- using `aria-label` or `aria-labelledby` based on use case (if Switch doesn't have any children...)

### Problem Statement/Justification

With current DOM structure (label tag wrapping the whole content) we recently noticed a small accessibility concern, about accessible name.

Indeed, the component itself, identified through the `role="switch"` attribute, doesn't have any accessible name. This is due to component text (aka children) added as a sibling DOM node, and so not directly related. This can be confirmed using the "Axe" Chrome accessibility extension, and testing the component.

![image](https://github.com/adevinta/spark/assets/66770550/fbe178d2-8dcc-43f4-8913-031e0d228c92)

### Proposed Solution or API

After discussions with @andresz1, @acd02 and @Powerplex we agreed:
- changing the DOM structure, to match the Radix pattern: having the label as a sibling, linked through an `htmlFor` attribute. The whole content wrapped within a neutral div, for ex.
- using `aria-label` or `aria-labelledby` based on use case (if Switch doesn't have any children...)

```tsx
import React from 'react';
import * as Switch from '@radix-ui/react-switch';

const SwitchDemo = () => (
  <form>
    <div>
      <label htmlFor="airplane-mode">
        Airplane mode
      </label>
      <Switch.Root id="airplane-mode">
        <Switch.Thumb />
      </Switch.Root>
    </div>
  </form>
);

export default SwitchDemo;
```
